argilla/src/argilla/records/_resource.py

```diff
@@ -258,27 +262,62 @@ def from_model(cls, model: RecordModel, dataset: "Dataset") -> "Record":
         )
 
 
-class RecordFields:
+class RecordFields(dict):
     """This is a container class for the fields of a Record.
-    It allows for accessing fields by attribute and iterating over them.
+    It allows for accessing fields by attribute and key name.
     """
 
-    def __init__(self, fields: Dict[str, Union[str, None]]) -> None:
-        self.__fields = fields or {}
-        for key, value in self.__fields.items():
-            setattr(self, key, value)
+    def __init__(self, fields: Optional[Dict[str, FieldValue]] = None) -> None:
+        super().__init__(fields or {})
 
-    def __getitem__(self, key: str) -> Optional[str]:
-        return self.__fields.get(key)
+    def __getattr__(self, item: str):
+        return self[item]
 
-    def __iter__(self):
-        return iter(self.__fields)
+    def __setattr__(self, key: str, value: MetadataValue):
+        self[key] = value
 
-    def to_dict(self) -> Dict[str, Union[str, None]]:
-        return self.__fields
+    def to_dict(self) -> dict:
+        return {key: value for key, value in self.items()}
 
-    def __repr__(self) -> str:
-        return self.to_dict().__repr__()
+
+class RecordMetadata(dict):
+    """This is a container class for the metadata of a Record."""
+
+    def __init__(self, metadata: Optional[Dict[str, MetadataValue]] = None) -> None:
+        super().__init__(metadata or {})
+
+    def __getattr__(self, item: str):
+        return self[item]
+
+    def __setattr__(self, key: str, value: MetadataValue):
+        self[key] = value
+
+    def to_dict(self) -> dict:
+        return {key: value for key, value in self.items()}
+
+    def api_models(self) -> List[MetadataModel]:
+        return [MetadataModel(name=key, value=value) for key, value in self.items()]
+
+
+class RecordVectors(dict):
+    """This is a container class for the vectors of a Record.
+    It allows for accessing suggestions by attribute and key name.
+    """
+
+    def __init__(self, vectors: Dict[str, VectorValue]) -> None:
+        super().__init__(vectors or {})
+
+    def __getattr__(self, item: str):
+        return self[item]
+
+    def __setattr__(self, key: str, value: VectorValue):
+        self[key] = value
+
+    def api_models(self) -> List[VectorModel]:
+        return [Vector(name=name, values=value).api_model() for name, value in self.items()]
+
+    def to_dict(self) -> Dict[str, List[float]]:
+        return {key: value for key, value in self.items()}
 
 
 class RecordResponses(Iterable[Response]):
```

Comment:
How will this class be used when users want to:
- add a first response after retrieving records?
- add subsequent responses after retrieving records?

Ideally, these should just be the same.